File: Level 2/2 x n 타일링/solution.js
function solution(n) {
  if (n === 1) return 1;
  if (n === 2) return 2;

  let a = 1;
  let b = 2;
  let target;

  for (let i = 3; i <= n; i++) {
    target = (a + b) % 1_000_000_007;
    a = b;
    b = target;
  }

  return target;
}

File: Level 2/2 x n 타일링/solution.py
def solution(n):
    if n == 1:
        return 1

    if n == 2:
        return 2

    dp = [0] * (n + 1)
    dp[1] = 1
    dp[2] = 2

    for i in range(3, n + 1):
        dp[i] = (dp[i - 2] + dp[i - 1]) % 1_000_000_007

    return dp[n]
